api/src/main/java/me/lokka30/treasury/api/economy/currency/Currency.java

```diff
@@ -12,69 +12,224 @@
+     * @return new currency instance
+     */
+    @NotNull
+    public static Currency of(
+            @NotNull UUID currencyId,
+            @Nullable Character currencyChar,
+            int roundedDigits,
+            double conversionCoefficient,
+            @Nullable Function<UUID, Double> startingBalance,
+            @NotNull BiFunction<Double, Locale, String> balanceFormatter,
+            @NotNull String... names
+    ) {
+        Objects.requireNonNull(currencyId, "currencyId");
+        return new Currency(
+                currencyId,
+                currencyChar,
+                roundedDigits,
+                conversionCoefficient,
+                startingBalance,
+                balanceFormatter,
+                names
+        );
+    }
+
+    private final UUID currencyId;
+    private final Character currencyChar;
+    private final String[] names;
+    private final int roundedDigits;
+    private final double conversionCoefficient;
+    private final Function<UUID, Double> startingBalance;
+    private final BiFunction<Double, Locale, String> balanceFormatter;
+
+    private Currency(@Nullable UUID currencyId,
+                     @Nullable Character currencyChar,
+                     int roundedDigits,
+                     double conversionCoefficient,
+                     @Nullable Function<UUID, Double> startingBalance,
+                     @NotNull BiFunction<Double, Locale, String> balanceFormatter,
+                     @NotNull String... names
+    ) {
+        if (currencyId == null) {
+            this.currencyId = UUID.randomUUID();
+        } else {
+            this.currencyId = currencyId;
+        }
+        this.currencyChar = currencyChar;
+        this.roundedDigits = roundedDigits;
+        this.conversionCoefficient = conversionCoefficient;
+        if (startingBalance == null) {
+            this.startingBalance = ($) -> 0d;
+        } else {
+            this.startingBalance = startingBalance;
+        }
+        if (names.length < 1) {
+            throw new IllegalArgumentException("Empty array specified for currency name.");
+        }
+        this.balanceFormatter = Objects.requireNonNull(balanceFormatter, "balanceFormatter");
+        this.names = Objects.requireNonNull(names, "names");
+    }
 
     /**
-     * @author lokka30
+     * Gets the {@link UUID} of the currency.
+     *
+     * @return the UUID of the currency
+     * @author lokka30, MrIvanPlays
      * @since v1.0.0
-     * Get the UUID of the currency.
-     * @return the UUID of the currency.
      */
     @NotNull
-    UUID getCurrencyId();
+    public UUID getCurrencyId() {
+        return currencyId;
+    }
+
+    /**
+     * Gets the character which identifies this currency, which may be null.
+     *
+     * @return char identifier. could be null
+     * @author MrIvanPlays
+     * @since v1.0.0
+     */
+    @Nullable
+    public Character getCurrencyCharacter() {
```

Comment:
Maybe this should be a String? Maybe the currency symbol has several characters

Reply:
it is not necessary to specify it anyway so it is staying as `char` due to that the most used global currencies are using characters (e.g. '$' is the dollar, '€' is the euro)

Reply:
I suppose that if someone wants to use a mylti character symbol he can just use a name